pkg/cache/task.go

```diff
@@ -196,6 +197,37 @@ func (task *Task) handleSubmitTaskEvent(event *fsm.Event) {
 		v1.EventTypeNormal, "TaskSubmitted",
 		"application \"%s\" task \"%s\" is submitted to the scheduler",
 		task.applicationID, task.taskID)
+
+	// after a small amount of time, if the task is still not able to get allocation,
+	// put the pod to unscheduable state. This will trigger the cluster-auto-scaler to launch
+	// the auto-scaling process.
+	// TODO improve this
+	// Note, this approach is suboptimal, when a task cannot be allocated, it doesn't always
+	// mean it needs the nodes to scale up. E.g task runs out of max capacity of the queue,
+	// user/app runs out of the limit etc. Ideally, we should add more interactions between
+	// core and shim to negotiate on when to set the state to unscheduable and trigger the
+	// auto-scaling appropriately.
+	go func(self *Task) {
+		time.Sleep(5*time.Second)
+		log.Logger.Info("updating pod state",
```

Comment:
The log comment should be checking pod state, we might not update at all.

Reply:
Moved the log to the inside of "if". And added more info.